openedx/core/djangoapps/content/learning_sequences/migrations/0014_remove_user_partition_group_duplicates.py

```diff
@@ -0,0 +1,33 @@
+# Generated by Django 2.2.24 on 2021-07-07 18:34
+
+from django.db import migrations
+from django.db.models import Count, Min
+
+
+class Migration(migrations.Migration):
+
+    def remove_user_partition_group_duplicates(apps, schema_editor):
+        UserPartitionGroup = apps.get_model('learning_sequences', 'UserPartitionGroup')
+
+        upg_duplicates = (
+            UserPartitionGroup.objects.values("partition_id", "group_id")
+            .annotate(first_entry_id=Min("id"), num_entries=Count("id"))
+            .filter(num_entries__gt=1)
+        )
+
+        # Delete all duplicates, excluding the original user partition group for the partition/group combo.
+        for duplicate in upg_duplicates:
+            (
+                UserPartitionGroup.objects
+                .filter(partition_id=duplicate["partition_id"], group_id=duplicate["group_id"])
+                .exclude(id=duplicate["first_entry_id"])
+                .delete()
+            )
+
```

Comment:
Does there need to be an update to move all the entries that pointed to the duplicate to point at the original?